mlflow/store/tracking/rest_store.py

```diff
@@ -1284,6 +1314,17 @@ def log_spans(self, experiment_id: str, spans: list[Span]) -> list[Span]:
         if not spans:
             return []
 
+        server_version = self._get_server_version(self.get_host_creds())
+        if not server_version:
+            raise NotImplementedError(
+                "log_spans is not supported: could not identify MLflow server version"
+            )
+        elif server_version < Version("3.4"):
+            raise NotImplementedError(
+                f"log_spans is not supported: MLflow server version {server_version} is"
+                f" less than 3.4"
+            )
```

Comment:
@harupy I considered an alternative approach of having the backend try calling `log_spans()`, but this doesn't actually work because existing backend code has no handlers for the `log_spans()`. Using the version number, which can already be retrieved from the server, seems like a clean approach.